scripts/integration/docker-compose.http-scrape.yml

```diff
@@ -0,0 +1,75 @@
+version: "3"
+
+services:
+  # https://github.com/sigoden/dufs
+  # serves static files at an HTTP endpoint
+  dufs:
+    image: docker.io/sigoden/dufs:latest
+    networks:
+      - backend
+    command:
+      - "/data"
+    volumes:
+      - ${PWD}/tests/data/http-scrape/:/data
+  # To validate Basic HTTP authentication option
+  dufs-auth:
+    image: docker.io/sigoden/dufs:latest
+    networks:
+      - backend
+    command:
+      - "-a"
+      - "/@user:pass"
+      - "--auth-method"
+      - "basic"
+      - "/data"
+    volumes:
+      - ${PWD}/tests/data/http-scrape/:/data
+  # To validate TLS options
+  dufs-https:
+    image: docker.io/sigoden/dufs:latest
+    networks:
+      - backend
+    command:
+      - "--tls-cert"
+      - "/certs/ca.cert.pem"
+      - "--tls-key"
+      - "/certs/ca.key.pem"
+      - "/data"
+    volumes:
+      - ${PWD}/tests/data/http-scrape/:/data
+      - ${PWD}/tests/data/ca/certs/ca.cert.pem:/certs/ca.cert.pem
+      - ${PWD}/tests/data/ca/private/ca.key.pem:/certs/ca.key.pem
```

Comment:
I think you want to use the localhost cert here, `tests/data/ca/intermediate_server/certs/localhost.cert.pem` (or possibly the one with the chain: `localhost-chain.cert.pem`).

The cert you have configured is the root CA cert which, I think, won't be valid for use as a server certificate due to the configured key usages.

Reply:
Hmmm. I verified that the `tests/data/ca/intermediate_server/certs/localhost.cert.pem` (or chain) and `ests/data/ca/intermediate_server/private/localhost.key` , checksums match. Yet still unable to get this to take. As a first step I'm doing locally in my ec2 instance to avoid any docker issues. And the error I keep getting is that the certificate is self-signed. I was trying to find other examples in the project using these files but I'm only finding unit tests... It almost sounds like the ca cert file needs to be installed in the certificate chain of the host , which I also tried on the ec2 but still no luck :/

Reply:
Here's one test that verifies the certificates (both server and client):

https://github.com/vectordotdev/vector/blob/b5fa9271aeeab6ccfd3957c40fc737f041902bcb/src/sources/socket/mod.rs#L426-L488

Generally you'll need to configure the `tls` options of the component to use the ca file you found as `ca_file`. Happy to take a look at this together too. Certificates can be finicky.

Reply:
Ah thanks! I'll have a go at it and let you know 👍